webviz_subsurface/plugins/_simulation_time_series/types/derived_ensemble_vectors_accessor_impl.py

```diff
@@ -0,0 +1,188 @@
+            get_selected_expressions(expressions, vectors)
+            if expressions is not None
+            else []
+        )
+        self._resampling_frequency = (
+            resampling_frequency if self._provider.supports_resampling() else None
+        )
+
+    def has_provider_vectors(self) -> bool:
+        return len(self._provider_vectors) > 0
+
+    def has_interval_and_average_vectors(self) -> bool:
+        return len(self._interval_and_average_vectors) > 0
+
+    def has_vector_calculator_expressions(self) -> bool:
+        return len(self._vector_calculator_expressions) > 0
+
+    def get_provider_vectors_df(
+        self, realizations: Optional[Sequence[int]] = None
+    ) -> pd.DataFrame:
+        """Get dataframe for the selected provider vectors"""
+        if not self.has_provider_vectors():
+            raise ValueError(
+                f'Vector data handler for provider "{self._name}" has no provider vectors'
+            )
+        return self._provider.get_vectors_df(
+            self._provider_vectors, self._resampling_frequency, realizations
+        )
+
+    def create_interval_and_average_vectors_df(
+        self,
+        realizations: Optional[Sequence[int]] = None,
+    ) -> pd.DataFrame:
+        """Get dataframe with interval delta and average rate vector data for provided vectors.
+
+        The returned dataframe contains columns with name of vector and corresponding interval delta
+        or average rate data.
+
+        Interval delta and average rate date is calculated with same sampling frequency as provider
+        is set with. I.e. resampling frequency is given for providers supporting resampling,
+        otherwise sampling frequency is fixed.
+
+        `Input:`
+        * realizations: Sequence[int] - Sequency of realization numbers to include in calculation
+
+        `Output:`
+        * dataframe with interval  vector names in columns and their cumulative data in rows.
+        `Columns` in dataframe: ["DATE", "REAL", vector1, ..., vectorN]
+
+        ---------------------
+        `TODO:`
+        * Verify calculation of cumulative
+        * IMPROVE FUNCTION NAME!
+        * See TODO in calculate_from_resampled_cumulative_vectors_df()
+        """
+        if not self.has_interval_and_average_vectors():
+            raise ValueError(
+                f'Vector data handler for provider "{self._name}" has no interval delta '
+                "and average rate vector names"
+            )
+
+        cumulative_vector_names = [
+            get_cumulative_vector_name(elm)
+            for elm in self._interval_and_average_vectors
+            if is_interval_or_average_vector(elm)
+        ]
+        cumulative_vector_names = list(sorted(set(cumulative_vector_names)))
+
+        # TODO: Fetch vectors df with correct sampling and perform calculation.
+        # Ensure valid sampling and ensure correct AVG_ calculation (unit/day)
+        # calculation, i.e num days between sampling points
```

Comment:
Issue?

Reply:
Should rather remove the comment, as calculation of AVG_ and INTVL_ now follows the same resampling as providers. Previously the calculation had its own frequency setting (as the original time series plugin). As AVG_ is [unit/day], the coming work would be to handle calculation if raw data is included - due to inconsistent sampling frequency and frequency which can be higher than one sample per day.